master/internal/api_project.go

```diff
@@ -214,7 +214,32 @@ func (a *apiServer) getProjectColumnsByID(
 	}
 
 	summaryMetricsSet := make(map[string]struct{})
+	validationMetricsSet := make(map[string]struct{})
 	for _, trial := range summaryMetrics {
+		if rawMetrics, ok := trial.SummaryMetrics["validation_metrics"]; ok {
```

Comment:
how hard is it to make a struct for these validation metrics and just unmarshal them rather than all this untyped `map[string]interface{}` stuff. i think that would help the code a lot.

Reply:
I'm not sure about using unmarshal because `SummaryMetrics["validation_metrics"]` uses dynamic keys... But I changed sql to calculate metrics name and type directly

Reply:
i meant even a struct that had `ValidationMetrics map[string]interface{}` or something, but i will look at what you did